### ISSUE TITLE
Build fails on musl based systems

### ISSUE THREAD

**Issue by ifreund** [ifreund/river#18]
```
./zig-cache/o/lLwaO1ZvYa5hBuplK9Yu4M1VOyALWCVjxTPNh2mP5Civhg28EzQVimsSKAc5GFDg/cimport.zig:5738:5: error: opaque types have unknown size and therefore cannot be directly embedded in structs
    last_event: struct_timespec,
    ^
./zig-cache/o/lLwaO1ZvYa5hBuplK9Yu4M1VOyALWCVjxTPNh2mP5Civhg28EzQVimsSKAc5GFDg/cimport.zig:6516:5: note: while checking this field
    seat: [*c]struct_wlr_seat,
    ^
./zig-cache/o/lLwaO1ZvYa5hBuplK9Yu4M1VOyALWCVjxTPNh2mP5Civhg28EzQVimsSKAc5GFDg/cimport.zig:6476:5: note: while checking this field
    popup: [*c]struct_wlr_xdg_popup,
    ^
./zig-cache/o/lLwaO1ZvYa5hBuplK9Yu4M1VOyALWCVjxTPNh2mP5Civhg28EzQVimsSKAc5GFDg/cimport.zig:6493:5: note: while checking this field
    unnamed_161: union_unnamed_162,
    ^
./zig-cache/o/lLwaO1ZvYa5hBuplK9Yu4M1VOyALWCVjxTPNh2mP5Civhg28EzQVimsSKAc5GFDg/cimport.zig:6617:5: note: while checking this field
    surface: [*c]struct_wlr_xdg_surface,
    ^
./src/decoration.zig:11:5: note: while checking this field
    wlr_xdg_toplevel_decoration: *c.wlr_xdg_toplevel_decoration_v1,
    ^
/usr/lib/zig/std/linked_list.zig:22:13: note: while checking this field
            data: T,
            ^
/usr/lib/zig/std/linked_list.zig:52:9: note: while checking this field
        first: ?*Node,
        ^
./src/decoration_manager.zig:14:5: note: while checking this field
    decorations: std.SinglyLinkedList(Decoration),
    ^
./src/server.zig:27:5: note: while checking this field
    decoration_manager: DecorationManager,
    ^
./src/main.zig:13:26: note: referenced here
    var server: Server = undefined;
                         ^
river...The following command exited with error code 1:
/usr/bin/zig build-exe /root/river/src/main.zig --c-source -std=c99 /root/river/include/bindings.c --library c --library wayland-server --library wlroots --library xkbcommon --cache-dir /root/river/zig-cache --name river -target x86_64-linux-musl -I /root/river/protocol -I /root/river -I /usr/include -I /usr/include/libdrm -I /usr/include/pixman-1 -I /usr/include/elogind --cache on 

Build failed. The following command failed:
/root/river/zig-cache/o/3ycZuz4w9FPFWUv5IEQkCiCh7aamLMMWOpfPulKCPUqfJIj4VYEaYed9tOqnRpQJ/build /usr/bin/zig /root/river /root/river/zig-cache -Dtarget=x86_64-linux-musl
```

**Comment by ifreund** [comment]
Seems that this is likely blocked by https://github.com/ziglang/zig/issues/1499

For [glibc](https://github.com/ziglang/zig/blob/15141d865abda09c09d8f6e3385f127527c5232e/lib/libc/glibc/time/bits/types/struct_timespec.h) it translates somehow, but for [musl](https://github.com/ziglang/zig/blob/15141d865abda09c09d8f6e3385f127527c5232e/lib/libc/include/x86_64-linux-musl/bits/alltypes.h#L229) it fails.

**Comment by ifreund** [comment]
Attempted a workaround on this branch https://github.com/ifreund/river/tree/musl-timespec-hack. However, river segfaults on starting up so something is going very wrong. I'm putting work on this issue on hold for now to focus on other things.

**Comment by ifreund** [comment]
The current plan for solving this is to write custom zig bindings for wlroots which use the `timespec` type from the zig standard library, avoiding the issue. Before working on idiomatic zig wlroots bindings I want to first create idiomatic zig wayland bindings however, which I've started preliminary work on.

**Comment by voroskoi** [comment]
Thank You for the heads up!

**Comment by maringuu** [comment]
A workaround for everyone developing river on alpine:
1. Create a chroot with a glibc based distro (e.g. Arch)
2. Compile river in the chroot
3. Run river on alpine using [gcompat](https://code.foxkit.us/adelie/gcompat)
4. To get gdb working as exepected use `patchelf --set-interpreter /lib/ld-musl-x86_64.so.1`

**Comment by ifreund** [comment]
This should be fixed by #159 but I haven't got a musl system set up to check currently
